rtt_rosparam/include/rtt_rosparam/ros_param_data_source.hpp

```diff
@@ -115,15 +112,22 @@ class RosParamDataSource
   void set( typename internal::AssignableDataSource<T>::param_t t )
   {
     set() = t;
-    ros::param::set(mparam_name_, t);
+    updated();
   }
 
-  // There is not referred element, no allocation exists for this data source and it is not an alias.
+  // There is not referred element, no allocation exists for this data source
+  // and it is not an alias.
+  // After working with the reference, if something was assigned, the data
+  // source may need a call to updated() after set()
   typename internal::AssignableDataSource<T>::reference_t set()
   {
     return mcached_data_;
   }
 
+  void updated() {
+    ros::param::set(mparam_name_, mcached_data_);
```

Comment:
nit: style consistency
```suggestion
  void updated()
  {
    ros::param::set(mparam_name_, mcached_data_);
```